Parse/src/main/java/com/parse/ParseFile.java

```diff
@@ -273,12 +335,21 @@ public void save() throws ParseException {
           return Task.cancelled();
         }
 
-        return getFileController().saveAsync(
-            state,
-            data,
-            sessionToken,
-            progressCallbackOnMainThread(uploadProgressCallback),
-            cancellationToken).onSuccessTask(new Continuation<State, Task<Void>>() {
+        Task<ParseFile.State> saveTask = data != null ?
+            getFileController().saveAsync(
+                state,
+                data,
+                sessionToken,
+                progressCallbackOnMainThread(uploadProgressCallback),
+                cancellationToken) :
+            getFileController().saveAsync(
+                state,
+                file,
+                sessionToken,
+                progressCallbackOnMainThread(uploadProgressCallback),
+                cancellationToken);
```

Comment:
nit: Can we just use a `if`/`else` here instead? This ternary operator is weird to look at